mathesar_ui/src/systems/table-view/table-inspector/record/RowActions.svelte

```diff
@@ -26,17 +27,19 @@
 
   async function handleDeleteRecords() {
     if (!isDeleting) {
-      try {
-        isDeleting = true;
-        selection.freezeSelection = true;
-        await recordsData.deleteSelected(selectedRowIndices);
-        selection.resetSelection();
-      } catch (e) {
-        toast.fromError(e);
-      } finally {
-        selection.freezeSelection = false;
-        isDeleting = true;
-      }
+      isDeleting = true;
```

Comment:
No need to maintain a boolean for the loading state. The `confirmDelete` modals/popups already take care by closing after the promise is resolved.

Reply:
The loading icon of the `Delete` button depends on the `isDeleting` variable. Should, I go ahead and set it to `true` so it does not show the Loading icon during deletion ?

Reply:
i meant `false`.

Reply:
Yes @Hatilar420 that loader is not required since the main CTA on the confirmation modal shows the loader.

Reply:
Understood. will change it.
Thanks for the help :)